### ISSUE TITLE
FlyMine: create-utr-references is failing

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/intermine#1153]
This was on the protein interactions branch, not sure if this is also an issue on the beta branch.

```
BUILD FAILED
/data/code/intermine/imbuild/postprocess.xml:31: error running PostProcessTask (action: create-utr-references)
        at org.intermine.task.PostProcessTask.doCorePostProcess(PostProcessTask.java:148)
        at org.intermine.task.PostProcessTask.execute(PostProcessTask.java:121)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
        at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        at org.apache.tools.ant.Task.perform(Task.java:348)
        at org.apache.tools.ant.Target.execute(Target.java:390)
        at org.apache.tools.ant.Target.performTasks(Target.java:411)
        at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
        at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
        at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
        at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
        at org.apache.tools.ant.Main.runBuild(Main.java:809)
        at org.apache.tools.ant.Main.startAnt(Main.java:217)
        at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
        at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.intermine.task.PostProcessTask.doCorePostProcess(PostProcessTask.java:146)
        ... 17 more
Caused by: Operation was:create-utr-references
        at org.intermine.bio.postprocess.PostProcessOperationsTask.execute(PostProcessOperationsTask.java:232)
        ... 22 more
Caused by: java.lang.IllegalArgumentException: Couldn't set field "Transcript.fivePrimeUTR" to "FivePrimeUTR [chromosome=1000092, chromosomeLocation=72110596, cytoLocation=null, gene=null, id=72110597, length=22, name=null, organism=1000001, primaryIdentifier="FBtr0071388-5-prime-utr", score=null, scoreType=null, secondaryIdentifier=null, sequence=72110598, sequenceOntologyTerm=null, symbol=null]" (a org.intermine.model.bio.FivePrimeUTRShadow)
        at org.intermine.util.DynamicUtil.setFieldValue(DynamicUtil.java:164)
        at org.intermine.model.bio.TranscriptShadow.setFieldValue(TranscriptShadow.java:508)
        at org.intermine.bio.postprocess.CreateReferences.createUtrRefs(CreateReferences.java:405)
        at org.intermine.bio.postprocess.PostProcessOperationsTask.execute(PostProcessOperationsTask.java:120)
        ... 22 more
Caused by: java.lang.NullPointerException
        at org.intermine.util.DynamicUtil.setFieldValue(DynamicUtil.java:154)
        ... 25 more

Total time: 26 seconds
``

**Comment by julie-sullivan** [comment]
@sergiocontrino None of the code has changed recently, besides the dynamic object foolishness #1123 so that is the likely culprit.

Some idiot had stupidly removed the test for UTRs. !!! 3aac2f17416aff4ccab7fd7bf40e1ddc4e1a65b0

I reverted this dumb change and the UTR test passed but another test failed. Look familiar?

```
   at org.intermine.bio.postprocess.CreateReferencesTest.testInsertReferenceField(CreateReferencesTest.java:101)
   Caused by: org.intermine.objectstore.ObjectStoreException: Error - could not fetch object with ID of 3001 for query SELECT DISTINCT a1_ FROM org.intermine.model.bio.Gene AS a1_ WHERE a1_.primaryIdentifier = 'gene1'
   at org.intermine.objectstore.intermine.ResultsConverter.convert(ResultsConverter.java:204)
   at org.intermine.objectstore.intermine.ObjectStoreInterMineImpl.executeWithConnection(ObjectStoreInterMineImpl.java:1236)
   at org.intermine.objectstore.intermine.ObjectStoreInterMineImpl.executeWithConnection(ObjectStoreInterMineImpl.java:1131)
   at org.intermine.objectstore.intermine.ObjectStoreInterMineImpl.execute(ObjectStoreInterMineImpl.java:1022)
   at org.intermine.objectstore.query.ResultsBatches.fetchBatchFromObjectStore(ResultsBatches.java:192)
   at org.intermine.objectstore.query.PrefetchManager.doRequest(PrefetchManager.java:241)
   at org.intermine.objectstore.query.PrefetchManager.doRequest(PrefetchManager.java:146)

```

Is this related to #1150?

**Comment by julie-sullivan** [comment]
That postprocess is working correctly after all. The data is wrong somehow.

OLD

```
flymine-beta-2015-aug-07=# select primaryidentifier,class from transcript where primaryidentifier = 'FBtr0071388';
 primaryidentifier |            class             
-------------------+------------------------------
 FBtr0071388       | org.intermine.model.bio.MRNA
(1 row)
```
NEW

```
flymine-build=# select primaryidentifier,class from transcript where primaryidentifier = 'FBtr0071388';
 primaryidentifier |               class                
-------------------+------------------------------------
 FBtr0071388       | org.intermine.model.bio.Transcript
(1 row)
```

**Comment by julie-sullivan** [comment]
There are lots of subclasses in the transcript table. Why is this transcript so different?

```
flymine-build=# select class, count(*) from transcript group by class;
               class                | count  
------------------------------------+--------
 org.intermine.model.bio.SnoRNA     |   2536
 org.intermine.model.bio.MiRNA      |   1572
 org.intermine.model.bio.NcRNA      |  11068
 org.intermine.model.bio.RRNA       |   1300
 org.intermine.model.bio.TRNA       |   3492
 org.intermine.model.bio.MRNA       | 408453
 org.intermine.model.bio.SnRNA      |    388
 org.intermine.model.bio.Transcript |  42312
(8 rows)
```

**Comment by julie-sullivan** [comment]
Sources of interest

```
flymine-build=# select s.sourcename from tracker s, transcript t where t.id = s.objectid and t.class = 'org.intermine.model.bio.Transcript' group by sourcename;
          sourcename           
-------------------------------
 anopheles-identifiers
 skel_chado-db-flybase-dmel
 flybase-dmel-cds-fasta
 affy-probes
 chado-db-flybase-dmel
 flybase-dmel-5prime-utr-fasta
 miranda
 flybase-dmel-3prime-utr-fasta
 long-oligo
```

**Comment by julie-sullivan** [comment]
#1096 - changed the data tracker
#1100  - remove multiclass dynamic objects

I ran full builds after each of these changes before merging them in
 though. Maybe I didn't update correctly.

**Comment by julie-sullivan** [comment]
I reverted #1100 and the flymine build worked fine. I thought I had tested that!! So I am going to revert this change with the idea we will fix this bug and reimplement. Hopefully this will fix our test failures too, similar symptoms I think.

**Comment by julie-sullivan** [comment]
Fixed by 0324adb182c526c2245417ec1118e787836bad85
